.prow/scripts/publish-java-sdk.sh

```diff
@@ -69,4 +69,4 @@ gpg --import --batch --yes $GPG_KEY_IMPORT_DIR/private-key
 echo "============================================================"
 echo "Deploying Java SDK with revision: $REVISION"
 echo "============================================================"
-mvn --projects sdk/java -Drevision=$REVISION --batch-mode clean deploy
+mvn --projects datatypes/java,sdk/java -Drevision=$REVISION --batch-mode clean deploy
```

Comment:
@davidheryanto I went with the bare minimum of augmenting the script here, since it makes sense that these are always published together by the same job, the conditions for when to run are the same, etc. Do you think anything more than this is needed?

We will need to forward-port this change to master, I forgot it in #391 since the release process came after I started.